comment-updater-core/src/main/kotlin/org/jetbrains/research/commentupdater/processors/ProjectMethodExtractor.kt

```diff
@@ -18,29 +23,55 @@ object ProjectMethodExtractor {
     fun extractChangedMethods(
         project: Project,
         change: Change,
-        refactorings: List<Refactoring>,
+        allRefactorings: List<Refactoring>,
+        fileRefactorings: List<Refactoring>,
         statisticContext: HashMap<String, Int> = hashMapOf()
-    ): MutableList<Pair<PsiMethod, PsiMethod>> {
-        val before = change.beforeRevision?.content ?: return mutableListOf()
+    ): MutableList<Triple<PsiMethod, PsiMethod, Boolean>> {
+        val before = change.beforeRevision?.content ?: ""
         val after = change.afterRevision?.content ?: return mutableListOf()
 
-        val renameMapping = RefactoringUtils.extractNameChanges(refactorings)
+        val renameMapping = RefactoringUtils.extractNameChanges(fileRefactorings)
 
-        val changedMethodPairs = mutableListOf<Pair<PsiMethod, PsiMethod>>()
+        val changedMethodPairs = mutableListOf<Triple<PsiMethod, PsiMethod, Boolean>>()
 
         val oldNamesToMethods = extractNamesToMethods(project, before, statisticContext)
 
         val newMethods = extractMethodsWithNames(project, after)
 
+
         newMethods.forEach { (afterName, newMethod) ->
             val beforeName = if (renameMapping.containsKey(afterName)) {
                 renameMapping[afterName]
             } else {
                 afterName
             }
-            oldNamesToMethods.get(beforeName)?.let { oldMethod ->
-                changedMethodPairs.add(oldMethod to newMethod)
+
+            if (!oldNamesToMethods.containsKey(beforeName)) {
+                var isNew = true
+
+                allRefactorings.filter { it.refactoringType == RefactoringType.MOVE_OPERATION || it.refactoringType == RefactoringType.MOVE_AND_RENAME_OPERATION }
+                    .forEach { ref ->
+                        val refactoring = (ref as MoveOperationRefactoring)
+                        val newFullName = refactoring.movedOperation.className + "." + refactoring.movedOperation.name
+
+                        var newMethodName = ""
+                        ApplicationManager.getApplication().runReadAction {
+                            newMethodName = newMethod.qualifiedName
+                        }
+
+                        if (newFullName == newMethodName) {
+                            isNew = false
+                        }
+
+                    }
+
+                changedMethodPairs.add(Triple(newMethod, newMethod, isNew))
```

Comment:
I think we can extract all methods from the previous commit to find an ancestor, but execution time will increase. Also, I use ```<newMethod, newMethod>``` because there is no implementation of empty or default PSIMethod.